addons/l10n_mx/models/account_account.py

```diff
@@ -17,3 +24,20 @@ def create(self, vals_list):
             tag_id = debit_tag.id if account.code[0] in DEBIT_CODES else credit_tag.id
             account.tag_ids = [Command.link(tag_id)]
         return accounts
+
+    @api.constrains('code', 'group_id')
+    def ensure_code_conforms_coa_sat(self):
+        incorrect_accounts = self.filtered(
+                lambda a: a.group_id and a.account_type != 'equity_unaffected' and a.company_id.country_code == 'MX'
+                and (a.code[:6] not in SAT_CODES or not THIRD_SECTION.fullmatch(a.code[6:]))
+        )
+        if incorrect_accounts:
+            account_names = ''.join(f'\n -   {a.name}' for a in incorrect_accounts)
+            raise RedirectWarning(
+                _("Some of your accounts do not respect the SAT code regulation.\n%s", account_names),
```

Comment:
```suggestion
                _("Some of your accounts do not respect the SAT code regulation.\n%(names_list)s", names_list=account_names),
```
Nitpick, but even if there's only one argument, I'd recommend using a named parameter